src/components/VmSnapshots/sagas.js

```diff
@@ -0,0 +1,50 @@
+import { takeEvery, put } from 'redux-saga/effects'
+
+import { ADD_VM_SNAPSHOT, DELETE_VM_SNAPSHOT, RESTORE_VM_SNAPSHOT } from './constants'
+import Api from '../../ovirtapi'
+import { callExternalAction, delay } from '../../saga/utils'
+import { fetchVmSnapshots, startProgress, stopProgress } from '../../sagas'
+import { addSnapshotRemovalPendingTask, removeSnapshotRemovalPendingTask } from '../../actions'
+
+function* addVmSnapshot (action) {
+  const snapshot = yield callExternalAction('addNewSnapshot', Api.addNewSnapshot, action)
+
+  if (snapshot && snapshot.id) {
+    yield fetchVmSnapshots({ vmId: action.payload.vmId })
+  }
+}
+
+function* deleteVmSnapshot (action) {
+  const snapshotId = action.payload.snapshotId
+  const vmId = action.payload.vmId
+  const result = yield callExternalAction('deleteVmSnapshot', Api.deleteVmSnapshot, { payload: { snapshotId, vmId } })
+  if (result.error) {
+    return
+  }
+  yield put(addSnapshotRemovalPendingTask(snapshotId))
+  let snapshotRemoved = false
+  for (let delaySec of [ 4, 4, 4, 30, 60 ]) {
+    const apiSnapshot = yield callExternalAction('snapshot', Api.snapshot, { payload: { snapshotId, vmId } }, true)
+    if (apiSnapshot.error && apiSnapshot.error.status === 404) {
+      snapshotRemoved = true
+      break
+    }
+    yield delay(delaySec * 1000)
+  }
+  if (snapshotRemoved) {
+    yield fetchVmSnapshots({ vmId })
+  }
+  yield put(removeSnapshotRemovalPendingTask(snapshotId))
```

Comment:
Do I understand correctly, that there's up to 4+4+4+30+60 seconds of no information about a pending task for the user before proper UI state changes? If so, it would be better to show a pending info immediately.

Reply:
Actually no, task is added to pending tasks, so user will see marked off list item and "alert" about pending task.